MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/QVectorsWidget.py

```diff
@@ -21,19 +21,43 @@
     QAbstractScrollArea,
     QComboBox,
     QDialog,
+    QGridLayout,
     QHBoxLayout,
     QLabel,
     QPushButton,
     QSizePolicy,
+    QSpinBox,
     QTableView,
+    QTabWidget,
     QVBoxLayout,
+    QWidget,
 )
 
+from MDANSE.Framework.Configurators.QVectorsConfigurator import QVectorsConfigurator
 from MDANSE.Framework.QVectors.IQVectors import IQVectors
+from MDANSE.MLogging import LOG
 from MDANSE_GUI.InputWidgets.WidgetBase import WidgetBase
 from MDANSE_GUI.Tabs.Models.PlottingContext import PlottingContext
-from MDANSE_GUI.Tabs.Views.PlotDataView import convert_vectors_to_datasets
+from MDANSE_GUI.Tabs.Views.PlotDataView import (
+    vector_angular_datasets,
+    vector_projection_datasets,
+    vector_q_statistics_datasets,
+)
 from MDANSE_GUI.Tabs.Visualisers.PlotWidget import PlotWidget
+from MDANSE_GUI.Utils import block_signals
+
+
+def numerator_suffix(index: int) -> str:
+    last_digit = int(str(index)[-1])
+    match last_digit:
+        case 1:
+            return "st"
+        case 2:
+            return "nd"
+        case 3:
+            return "rd"
+        case _:
+            return "th"
```

Comment:
Unless we expect `-N`th values.
```suggestion
    match index % 10:
        case 1:
            return "st"
        case 2:
            return "nd"
        case 3:
            return "rd"
        case _:
            return "th"
```

Reply:
I think I messed this function up anyway, because we should have:
1st
**11th**
21st
31st
...